### PR TITLE
Support fetching database credentials from AWS Secrets Manager.

### PR DESCRIPTION
**Pull request by rgooch**

### CODE REVIEW

File: cmd/keymasterd/storage.go [hunk 1]
```diff
@@ -20,6 +20,26 @@ const userProfileSuffix = ".gob"
...
+	}
+	metadataClient, err := getMetadataClient()
+	if err != nil {
+		return err
+	}
+	secrets, err := getAwsSecret(metadataClient, config.AwsSecretId)
+	if err != nil {
+		return err
+	}
+	config.StorageUrl = os.Expand(config.StorageUrl,
```
Comment (cviecco) [inline cmd/keymasterd/storage.go]: While this works I am not very happy on leaving the secrets on the environment. Any other way? Also you give users too much leeeway.

**Comment by cviecco** [comment]
Before final merge you need docs on how to use this feature. But do this last.

File: cmd/keymasterd/storage.go (hunk 1 above)
Comment (rgooch) [reply cmd/keymasterd/storage.go]: The secrets are not in the environment; they are only in memory. The `os.Expand()` function is just a variable expander.

**Comment by rgooch** [comment]
I see we have `docs/keymaster-openidc-idp.md`, but we're missing a lot of other configuration documentation. While I could write a simple file addressing this new capability, it might be better to include an example configuration file for AWS. Thoughts?

File: cmd/keymasterd/storage.go (hunk 1 above)
Comment (rgooch) [reply cmd/keymasterd/storage.go]: The approach I took gives users the flexibility to store the hostname of the Postgres server in AWS Secrets Manager too, if they wish, which I think is a valuable feature. Note that `os.Expand()` _does not_ pull from the environment.

**Comment by rgooch** [comment]
I have gone ahead and added a sample configuration file for AWS, with some comments.

File: cmd/keymasterd/storage.go (hunk 1 above)
Comment (cviecco) [reply cmd/keymasterd/storage.go]: I actually misread the docs.. I was reading at expandEnv.. not expand.

**Pull request merged**